aea/contracts/base.py

```diff
@@ -257,6 +257,42 @@ def build_transaction(
         )
         return tx
 
+    @classmethod
+    def default_method_call(
+        cls, ledger_api: LedgerApi, contract_address: str, **kwargs: Any
+    ) -> Optional[JSONLike]:
+        """
+        Make a contract call.
+
+        :param ledger_api: the ledger apis.
+        :param method_name: the contract method name.
+        :param kwargs: keyword arguments.
+        :return: the call result
+        """
+
+        contract_instance = cls.get_instance(ledger_api, contract_address)
+        # contract_instance = cls.get_instance(ledger_api)  # only for deployment txs
+
+        # Get the function building its signature from kwargs:
+
+        # Contract.get_function_by_signature(signature)
+        # -------------------------------------------------------------
+        # Searches for a distinct function with matching signature.
+        # Returns an instance of ContractFunction upon finding a match.
+        # Raises ValueError if no match is found.
+        # contract.get_function_by_signature('identity(uint256,bool)')
+
+        # Other options
+        # -------------
+        # Contract.find_functions_by_name(name), Contract.get_function_by_name(name)
+
+        method_name = "get_name_from_function_above"
```

Comment:
Basically build the function signature from kwargs, get its name and call it.

Reply:
yes - we already have this code elsewhere on open-autonomy or somewhere I think